### PR TITLE
fix: adding cosine distance and results to the benchmark

### PR DESCRIPTION
**Pull request by GabrielBeFr**
### What
- Adding results of benchmark ran with cosine distance
- Adding code for running the benchmark with cosine distance

### CODE REVIEW

File: logo-ann/benchmark/main.py [hunk 1]
```diff
@@ -122,6 +122,10 @@ def pairwise_squared_euclidian_distance(A: np.ndarray) -> torch.Tensor:
     squared_sum = torch.sum(A ** 2.0, axis=1, keepdim=True)
     return squared_sum + squared_sum.T - 2 * dot_product
 
+def cosine_similarity(A: torch.Tensor) -> torch.Tensor:
+    assert len(A.shape) == 2
+    normalized = torch.nn.functional.normalize(A,p=2.0,dim=1)
```
Comment (raphael0202) [inline logo-ann/benchmark/main.py]: Not important here (as it's research code), but it's good to follow PEP8:
`normalized = torch.nn.functional.normalize(A,p=2.0,dim=1)` -> `normalized = torch.nn.functional.normalize(A, p=2.0, dim=1)`

File: logo-ann/benchmark/main.py [hunk 2]
```diff
@@ -356,7 +360,7 @@ def run_latency_benchmark(device: torch.device):
         "benchmark", help="Launch latency benchmark"
     )
     benchmark_parser.add_argument("--gpu", action="store_true")
-    args = parser.parse_args()
+    args = parser.parse_args()#put "evaluate" or "benchmark" to run the right function
```
Comment (raphael0202) [inline logo-ann/benchmark/main.py]: Same PEP8 comment, there must be two spaces before an inline comment

File: logo-ann/benchmark/main.py [hunk 3]
```diff
@@ -304,7 +308,7 @@ def evaluate():
             # print(f"Labels shape: {labels.shape}")
             with torch.inference_mode():
                 distance_matrix = (
-                    pairwise_squared_euclidian_distance(embeddings).cpu().numpy()
+                    pairwise_squared_euclidian_distance(embeddings).cpu().numpy() #you can change here the distance used
```
Comment (raphael0202) [inline logo-ann/benchmark/main.py]: It would have been good to have an argparse option to change the distance used instead of updating it in the code

**Review by raphael0202 — APPROVED** [review]
